src/hooks/lib/preview/compounder/useTokenPreviewDepositFallback.ts

```diff
@@ -52,21 +65,110 @@ export default function useTokenPreviewDepositFallback({
     enabled: isUnderlyingAsset && enabled,
   })
 
-  if (isUnderlyingAsset && ybTokenSymbol === "fcGLP") {
-    return {
-      ...previewUnderlying,
-      data: {
-        minAmountWei: BigInt(previewUnderlying?.data ?? 0).toString(),
-        resultWei: BigInt(previewUnderlying?.data ?? 0).toString(),
-      },
-    }
+  return isUnderlyingAsset && ybTokenSymbol === "fcGLP"
+    ? {
+        ...previewUnderlying,
+        data: {
+          minAmountWei: BigInt(previewUnderlying?.data ?? 0).toString(),
+          resultWei: BigInt(previewUnderlying?.data ?? 0).toString(),
+        },
+      }
+    : {
+        ...preview,
+        data: {
+          minAmountWei: undefined,
+          resultWei: (preview.data ?? BigInt(0)).toString(),
+        },
+      }
+}
+
+function useTokenPreviewDepositUnderlying({
+  token,
+  amount,
+  slippage,
+  enabled,
+}: {
+  token?: Address
+  amount: string
+  slippage: number
+  enabled: boolean
+}) {
+  const chainId = useActiveChainId()
+  const isArbitrumFamily = chainId === 313371 || chainId === 42161
+  const [glpPrice, setGlpPrice] = useState(0)
+
+  token = token ?? "0x"
+  if (token == ETH) {
+    token = WETH_ARBI
   }
+  useEffect(() => {
+    getGlpPrice().then((price) => setGlpPrice(price))
+  }, [])
+
+  const tokenPriceWei = useContractRead({
+    chainId,
+    abi: GlpVault,
+    address: glpVault,
+    enabled: isArbitrumFamily && enabled,
+    functionName: "getMinPrice",
+    args: [token !== ETH ? token : WETH_ARBI],
+  })
+
+  const decimals: { [key: Address]: number } = {
```

Comment:
Move this outside/above the function; it's static data. It doesn't need to be a separate file. Just outside of the function.